### ISSUE TITLE
number of pairs after split and the ratio of negative SL samples

### ISSUE THREAD

**Issue by Karashan** [kunjiefan/MVGT-iSL#2]
Hi. I am a researcher interested in SL predictions and thanks for your inspiring work. I have two questions:

1. The train:val:test split was done at the gene level which with your code resulted in a very limited number of pairs entering the actual training and testing stage. At an example run of your code, I got the output as:
- number of train, val, test gene samples: 313 45 90
- number of train, val, test gene pair samples: **1492** 30 **134**
Can you confirm that the testing results in the paper are reported for just the ~134 pairs? Have your group tried separating the sets by pairs instead of genes so that train:val:test = 7:1:2?

2. In your code, I noticed that you specifically sampled the negative SL labels using a 1:1 ratio to match the number of positive samples, which makes the baseline of AUPR become ~0.5. This is not a problem but it would be interesting to see more experiments about increasing the ratio of negative samples (1:5, 1:20, 1:50, or just all). Have you tried a different positive-negative ratio and collected the results?

Thanks for reading. This is an interesting method and paper with a lot of work.

**Comment by kunjiefan** [comment]
Thanks for your interest in our work!

1. One of the innovation in our approach is using SL connectivity (node degree) rather than SL pair as the direct learning target, to avoid the learning of topology short-cut. To prove the power of our model in predicting novel (out-of-sample) genes, we make sure that genes in the test set are not seen during training, which result in the loss of some samples in the dataset. All genes included in 134 test gene pairs are not present in the training set, which is one way to show the generalization ability of the model. 
Previous methods have shown that it is easy to achieve high accuracy when datasets are separated randomly just by pairs (potentially overfitting and not useful in practice).

2. Yes, for all reported results in our paper, the test set is balanced, which to some extent might not reflect the true underlying distribution of SL, but it is fair to compare among all methods.

**Comment by Karashan** [comment]
Thanks for your clarification!
